### ISSUE TITLE
Reports of fatal error coming from plugin

### ISSUE THREAD

**Issue by arhine** [Parsely/wp-parsely#562]
Multiple Parse.ly users have reported the following error coming from the plugin:

```
Fatal error: Uncaught Error: gmdate() expects parameter 2 to be int, bool given
in /var/www/wp-content/plugins/wp-parsely/src/class-parsely.php on line 450
```

One shared data from its log files. I've obscured the customer name:

```
2021-12-27 14:08:23.821171 [NOTICE] [9118] [T0] [204.237.142.109:33881#APVH_xxxxxxx:443 (http://204.237.142.109:33881/#APVH_xxxxxxx:443) ] [STDERR] PHP Fatal error: Uncaught TypeError: gmdate() expects parameter 2 to be int, bool given in /var/www/vhosts/xxxxxxx/httpdocs/wp-content/plugins/wp-parsely/src/class-parsely.php:450

Stack trace:

#0 /var/www/vhosts/xxxxxxx/httpdocs/wp-content/plugins/wp-parsely/src/class-parsely.php(450): gmdate()

#1 /var/www/vhosts/xxxxxxx/httpdocs/wp-content/plugins/wp-parsely/src/class-parsely.php(189): Parsely\Parsely->construct_parsely_metadata()

#2 /var/www/vhosts/xxxxxxx/httpdocs/wp-includes/class-wp-hook.php(292): Parsely\Parsely->insert_page_header_metadata()

#3 /var/www/vhosts/xxxxxxx/httpdocs/wp-includes/class-wp-hook.php(316): WP_Hook->apply_filters()

#4 /var/www/vhosts/xxxxxxx/httpdocs/wp-includes/plugin.php(484): WP_Hook->do_action()

#5 /var/www/vhosts/xxxxxxx/httpdocs/wp-includes/general-template.php(3009): do_action()

#6 /var/www/vhosts/xxxxxxx/httpdocs/wp-content/themes/tpd-theme/header.php(18): wp_head()

#7 /var/www/vhosts/xxxxxxx/httpdocs/wp-includes/template.php(730): require_once('/var/www in /var/www/vhosts/xxxxxxx/httpdocs/wp-content/plugins/wp-parsely/src/class-parsely.php on line 450

When we de-activated the plugin, the site performance returned to normal levels.
```

**Comment by jblz** [comment]
Thanks for the report.

Given [the relevant line(s)](https://github.com/Parsely/wp-parsely/blob/a525caddeac334bfecb494f0d6b170615e8f967c/src/class-parsely.php#L450-L457) haven't changed in some time, it stands to reason it's the [recent addition](https://github.com/Parsely/wp-parsely/blob/a525caddeac334bfecb494f0d6b170615e8f967c/src/class-parsely.php#L9) of the [strict type declaration](https://www.php.net/manual/en/language.types.declarations.php) that's eliciting this behavior.

---

The [docs](https://developer.wordpress.org/reference/functions/get_post_time/) say `get_post_time` returns "False on failure" (same with `get_the_modified_date` & it is included in a `gmdate` call as well).

RE: "on failure," [the WordPress source](https://github.com/WordPress/WordPress/blob/5.8.2/wp-includes/general-template.php#L2699-L2708), shows `get_post_time` can return false in a couple of cases.

I guess some guards against passing non-int types to gmdate are in order.

**Comment by GaryJones** [comment]
Agreed on guard clauses. 

@arhine Can you please find out what type of content URLs these fatal errors were happening on? Post, page, term archive, author archive, home, etc? I know what the code/stacktrace says, but wondering if the runtime found its way into a logic branch for when the $post is empty.

**Comment by arhine** [comment]
@GaryJones I've heard back from one of the two customers, who reported they're only seeing it on post types thus far.

**Comment by arhine** [comment]
I've also encountered this issue on one of the sites I run, with a custom post type set to track as a post. Ping me if you need anything from me around that.

**Comment by arhine** [comment]
All -doesn't look like we're going to get much more in the way of feedback. The other customer that reported this got feedback from their hosting company that the plugin was causing issues, and they're still not really clear on where it was happening. I'm not having much luck getting more information.

Is there anything else that I can provide to help you resolve this?

**Comment by jblz** [comment]
> Is there anything else that I can provide to help you resolve this?

I tried to fidget with CPTs on a test site to trigger this failure, but wasn't able to. I don't think we really need to dig too much further into the failure cases...it's easy enough to simulate with this code in a plugin:

`add_filter( 'get_post_time', '__return_false' );`

**Comment by pauarge** [comment]
This should be fixed after merging #560 and #565. We will release this as a bug fix in 3.0.2. Thanks @arhine for the report!
